source/base/mpi.cc

```diff
@@ -324,12 +307,12 @@ namespace Utilities
                           "in a program since it initializes the MPI system."));
 
 
-
+      int ierr;
```

Comment:
We are reusing `ierr` for `PETScInitialize` in line 348 and hence this was not compiling using a non-MPI PETSc.
But this way `ierr` is unused if we use neither `MPI` nor `PETSc`. This is now fixed as well.